pkg/cmd/pulumi/logout.go

```diff
@@ -63,39 +61,31 @@ func newLogoutCmd() *cobra.Command {
 				cloudURL = "file://~"
 			}
 
-			if cloudURL == "" {
-				// Try to read the current project
-				project, _, err := readProject()
-				if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
-					return err
-				}
-
-				cloudURL, err = workspace.GetCurrentCloudURL(project)
-				if err != nil {
-					return fmt.Errorf("could not determine current cloud: %w", err)
-				}
-			}
-
-			var be backend.Backend
 			var err error
-			if filestate.IsFileStateBackendURL(cloudURL) {
-				fmt.Printf("Logged out of %s\n", cloudURL)
-				return workspace.DeleteAccount(cloudURL)
-			}
-
-			be, err = httpstate.New(cmdutil.Diag(), cloudURL, nil, workspace.GetCloudInsecure(cloudURL))
-			if err != nil {
-				return err
-			}
-
-			var logoutErr error
 			if all {
-				logoutErr = be.LogoutAll()
+				err = workspace.DeleteAllAccounts()
 			} else {
-				logoutErr = be.Logout()
+				if cloudURL == "" {
+					// Try to read the current project
+					project, _, err := readProject()
+					if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+						return err
+					}
+
+					cloudURL, err = workspace.GetCurrentCloudURL(project)
+					if err != nil {
+						return fmt.Errorf("could not determine current cloud: %w", err)
+					}
+
+					// Default to the default cloud URL. This means a `pulumi logout` will delete the
```

Comment:
This is the same behavior as before. If you trace the old code we would have called `httpstate.New` with a blank cloudURL and in that method we would have default to the pulumi.com url.